sticker-utils/src/subcommands/annotate.rs

```diff
@@ -0,0 +1,211 @@
+                    .help("Ignore sentences longer than N tokens"),
+            )
+            .arg(
+                Arg::with_name(READ_AHEAD)
+                    .help("Readahead (number of batches)")
+                    .long("readahead")
+                    .default_value("10"),
+            )
+    }
+
+    fn parse(matches: &ArgMatches) -> Self {
+        let config = matches.value_of(CONFIG).unwrap().into();
+        let batch_size = matches
+            .value_of(BATCH_SIZE)
+            .unwrap()
+            .parse()
+            .or_exit("Cannot parse batch size", 1);
+        let device = match matches.value_of("GPU") {
+            Some(gpu) => Device::Cuda(
+                gpu.parse()
+                    .or_exit(format!("Cannot parse GPU number ({})", gpu), 1),
+            ),
+            None => Device::Cpu,
+        };
+        let input = matches.value_of(INPUT).map(ToOwned::to_owned);
+        let max_len = matches
+            .value_of(MAX_LEN)
+            .map(|v| v.parse().or_exit("Cannot parse maximum sentence length", 1));
+        let output = matches.value_of(OUTPUT).map(ToOwned::to_owned);
+        let read_ahead = matches
+            .value_of(READ_AHEAD)
+            .unwrap()
+            .parse()
+            .or_exit("Cannot parse number of batches to read ahead", 1);
+
+        AnnotateApp {
+            batch_size,
+            config,
+            device,
+            input,
+            max_len,
+            output,
+            read_ahead,
+        }
+    }
+
+    fn run(&self) {
+        let config_file = File::open(&self.config).or_exit(
+            format!("Cannot open configuration file '{}'", &self.config),
+            1,
+        );
+        let mut config =
+            Config::from_toml_read(config_file).or_exit("Cannot parse configuration", 1);
+        config
+            .relativize_paths(&self.config)
+            .or_exit("Cannot relativize paths in configuration", 1);
+
+        let f = File::open(&config.labeler.labels).or_exit("Cannot open label file", 1);
+        let encoders: Encoders =
+            serde_yaml::from_reader(&f).or_exit("Cannot deserialize labels", 1);
+
+        for encoder in &*encoders {
+            eprintln!(
+                "Loaded labels for encoder '{}': {} labels",
+                encoder.name(),
+                encoder.encoder().len()
+            );
+        }
+
+        let tokenizer = config
+            .input
+            .word_piece_tokenizer()
+            .or_exit("Cannot read word pieces", 1);
+
+        let vectorizer = config
+            .input
+            .word_piece_vectorizer()
+            .or_exit("Cannot read word pieces", 1);
+
+        let bert_config = config
+            .model
+            .pretrain_config()
+            .or_exit("Cannot load pretraining model configuration", 1);
+
+        let mut vs = VarStore::new(self.device);
+
+        let model = BertModel::new(vs.root(), &bert_config, &encoders, 0.5)
```

Comment:
The dropout value of 0.5 has no effect?

Reply:
No, the models only use the dropout when the `train` argument to the model methods is `true`.

Reply:
Maybe it could become 0 here?

Reply:
0 would indeed be clearer. Fixed!